### ISSUE TITLE
Any plans to actually work on this add-on?  -general discussion about funding project

### ISSUE THREAD

**Issue by birdie-github** [lazarus-recovery/original_ff#2]
It no longer works in Firefox 53. Says: "I'm disabled".

**Comment by sierracircle** [comment]
We have been discussing  Firefox, and their plans to [discontinue XUL-based add-ons](https://blog.mozilla.org/addons/2015/08/21/the-future-of-developing-firefox-add-ons/) (which is what lazarus uses).

I personally have switched to [Palemoon Browser](http://www.palemoon.org), and plan on continuing the work on Lazarus, and adding it to the [Palemoon Add-ons site](https://addons.palemoon.org/).

Matt, the other guy working on this, sounds like he has grown dis-heartened by Firefox's plans and is re-evaluating what he wants to do.


Lazarus in its current form works fine in Palemoon. I have been working on removing the Donate nags and hope to have that completed soon.

**Comment by birdie-github** [comment]
Too bad lazarus v3 disappeared without a trace - I was under the impression that it wasn't based on XUL.

**Comment by sierracircle** [comment]
I made multiple attempts to contact the original developer, but never heard back.

Once this version is polished up a bit I might look into what it would take to migrate it to the new format...but at first glance it looks like there is no local storage allowed in the new add on design, which lazarus relies on to store your form entries...which means it would need to be online storage, which brings up a lot of security issues.

**Comment by birdie-github** [comment]
>  no local storage allowed in the new add on design, which lazarus relies on to store your form entries

This is certainly not true, otherwise WebExtensions couldn't allow to have any settings.

See `browser.storage.local.set()` and `browser.storage.local.get()`

https://developer.mozilla.org/Add-ons/WebExtensions/API/storage/local
https://developer.chrome.com/extensions/storage#property-local

**Comment by birdie-github** [comment]
Actually instead of porting an old add on, you could try reusing the Chrome's add-on: https://chrome.google.com/webstore/detail/lazarus-form-recovery/loljledaigphbcpfhfmgopdkppkifgno

It's barely usable right now, but it has most of the features already implemented so it's just the case of fixing its compatibility with modern Firefox.
[Lazarus-Form-Recovery_v3.0.6.zip](https://github.com/lazarus-recovery/original_ff/files/907560/Lazarus-Form-Recovery_v3.0.6.zip)

**Comment by sierracircle** [comment]
..are you interested in working on this project, or just making suggestions? I do not mean that in a snarky way..genuinely asking..

**Comment by birdie-github** [comment]
I'd be glad to do it all by myself but I literally know nothing about JavaScript and I'm not a programmer per se, so I hope you understand what I'm hinting at.

**Comment by sierracircle** [comment]
I am more of a plain-speaking type of person, so hints do not work so well on me.

**Comment by birdie-github** [comment]
Gotcha ;-)

Hints aside, I'd love to help, but I won't since my programming skills are too basic for this kind of work.

What I find really strange is that very few people use this absolutely indispensable add-on. Or maybe only few of them use Firefox beta/alpha, which means the public outcry is yet to begin.

**Comment by sierracircle** [comment]
I think that many people use the extension, but are not aware of the coming extension-apocalypse which looks like is planned for November. From Slashdot:
[Firefox 57, which will be preliminarily released on the 28th of Novermber, 2017, will only run WebExtensions: which means no XUL (overlay) add ons, no bootstrapped extensions, no SDK extensions and no Embedded WebExtensions. In other words by this date the chromification of Firefox will have been completed. If you depend on XUL add ons your only choice past this date will be Pale Moon.](https://tech.slashdot.org/story/17/02/17/1635216/mozilla-will-deprecate-xul-add-ons-before-the-end-of-2017)


Also, it is probably an intimidating thing to volunteer for if you are not a developer of any sort. I am a web-developer, so I have only some basic understanding, but I so far I have been able to muddle through enough.


We will see how it goes. For now, I am happy enough with Palemoon browser, and I am guessing Palemoon will get more support when November rolls around.

**Comment by birdie-github** [comment]
Palemoon in theory looks nice, in practice its web standards support is already light years behind all other major web browsers, and I don't see how this situation might change, considering that this browser has very few savvy developers. It will be impossible for Palemoon devs to sync their code base with modern Firefox because they've diverted too much.

**Comment by sierracircle** [comment]
Palemoon works for me, and solved a few issues I was having with Firefox crashing, plus Lazarus works fine in it, so I am not complaining at all..

**Comment by scytheria1001** [comment]
The plugin already stopped working in FF 53. I can't switch to a different browser because I use it for work. I found your comment on the Lazarus plugin page but it looks like you're not planning on trying to get Lazarus working on Firefox any more?

Installing TextCache caused FF to crash repeatedly for me so I had to uninstall it, so at this point there's no working alternative.

**Comment by sierracircle** [comment]
Hi.  We can certainly look into it; however, it is going to involve completely re-writing the plugin from top-to-bottom. First priority is cleaning up the existing plugin for those of us who have stuck with Palemoon or other branches of FF.

**Comment by nh2** [comment]
Can you make a donation page (for example Bountysource or Patreon) where people can donate to the effort of making Lazarus (or its functionality including storing the stored forms encrypted) work with the current and upcoming Firefox releases, and then advertise it big time?

I'd be in with 100 $.

**Comment by sierracircle** [comment]
I would be hesitant to take donations until I came up with an actual project outline for something like that so I could assess how much time/effort it would take.

I will consider working on a project outline this summer.

**Comment by guysoft** [comment]
I'd fund/suggest decently explained kickstarter/indiegogo.

**Comment by chickendude** [comment]
Over at [Reddit](https://www.reddit.com/r/firefox/comments/5nx8pk/lazarus_alternative/) a few months back some folks also discussed opening a bounty to fix Lazarus for the latest Firefox. I think this is something a lot of people would be willing to donate for if there was a real chance of it getting updated.

I'm not a professional JavaScript developer by any means, i just have some familiarity with it through using JS and jQuery in  websites, but i would be interested in helping make it compatible with the latest Firefox versions. Unfortunately i am not familiar at all with making extensions and have no idea what would be necessary. Lazarus is simply too useful to let it go...

**Comment by sierracircle** [comment]
@chickendude   In another week I will have some time to write up a project-outline, meaning studying and writing up what it would take to get the plugin converted to the new FF format (if it is even possible).

I feel like if we first explored it first we could better determine if it is even possible. You are welcome to co-write that with me. Once we have that, then we could discuss making a kickstarter or GoFundMe for the project.

I would not be comfortable taking donations until that document was written, in case it is not possible.  We might start writing and discover it is a simple task.

**Comment by chickendude** [comment]
@sierracircle Sure, i'd be happy to look into that with you (and browse through the current code to figure out what exactly it's doing).

**Comment by sierracircle** [comment]
@chickendude  great. I sent an invite to join the team.  I can start next week.

**Comment by chickendude** [comment]
@sierracircle Alright, cool, I'll talk to you then, then. :)

**Comment by dartraiden** [comment]
> at this point there's no working alternative
Try the [userscript](https://greasyfork.org/en/scripts/19509-textarea-backup-with-expiry-fix).

**Comment by franzalex** [comment]
@dartraiden Thanks for showing the userscript which offers a similar functionality.

I have modified it to capture and backup all `input` fields as well as include a right-click context menu.  

The new version is available [here](https://gist.github.com/franzalex/6bea78d9809ecb43886ad272cabff0ac).  
Direct URL to userscript: https://gist.github.com/franzalex/6bea78d9809ecb43886ad272cabff0ac/raw/TextArea_Backup_with_Expiry_Fix.user.js

**Comment by nh2** [comment]
Can you clarify, does this userscript encrypt the data and put it behind the Firefox master password?

**Comment by franzalex** [comment]
@nh2 No, it does not encrypt the captured data.

Though it may be possible to perform some sort of encryption, I don't think an encryption will be worth the effort as anyone with access to your modified script will have access to your encryption key. Also, I have not found GreaseMonkey can access Firefox's master password so I doubt that could be used to encrypt the entries.

**Comment by guysoft** [comment]
Can anyone explain where this comes from? And can it help us in any way?
https://chrome.google.com/webstore/detail/lazarus-form-recovery/loljledaigphbcpfhfmgopdkppkifgno?hl=en

**Comment by franzalex** [comment]
@guysoft I have no idea whether the extension you refer to was developed by the same person who made the version for Firefox.

Anyway, the discussion here concerns Firefox's add-on so let us limit ourselves to that unless there is something of interest specifically in the Chrome extension you'd like to point out.

**Comment by sierracircle** [comment]
@chickendude   I have been looking around but cannot find a way to PM members.  My email is adam@sierracircle.org   if you want to drop me a line.

**Comment by Tormen** [comment]
As there seems no password protected alternative ff addon,
and firefox built-in form-saver does not seem to do the job either [1], I would really love to see a current ff compatible version as well.

Willing to chip in money. Maybe willing to help with the code on limited aspects.

[1] https://www.berrange.com/posts/2011/06/22/firefox-form-data-history-a-goldmine-of-unencrypted-sensitive-personal-data/

**Comment by franzalex** [comment]
After doing a bit of research, I think I may have found a way to optionally encrypt the saved data.
It will, however, take some time before I get to implement this as I am currently busy working on another project.

I'll post an update to this once it's done! ;-)

> [@nh2](https://github.com/lazarus-recovery/original_ff/issues/2#issuecomment-309765432)
>
> Can you clarify, does this userscript encrypt the data and put it behind the Firefox master password?

**Comment by sNiXx** [comment]
Would also be willing to support this financially, as it was a very useful addon and textarea cache is not a proper replacement for me. Any news from the developers regarding progress and if there will be a WebExtension version?

**Comment by sierracircle** [comment]
I have not done much work on it, but I do have a stripped-down version running in PaleMoon Browser just fine.

It seems like there is enough general interest to do a project write-up (which is where we would determine the cost/time involved in converting it to webExtension)

From feebback so far, it sounds like converting it is the main interest.

I will do a write-up this September (2017) and we can go from there. Hang on to your funds until we have a realistic estimate.

**Comment by sierracircle** [comment]
Palemoon works for me, but I agree that it would be nice to have this working in the most recent FF.

I think we can all agree that having the features that Lazarus provides are indispensable.  More than once it has save my sanity by recovering particularly long forms that were lost on submit (especially when I was taking online classes and had to type all my work into online forms)

**Comment by chickendude** [comment]
I agree, Lazarus has saved me more times than i can count. I'm still happy to help work on porting(/figuring out how to port) this over.

**Comment by DarkVoyage** [comment]
Here's the latest version 3.2 for Firefox that was only available at getlazarus.com site. Maybe it helps in new development. Would be nice to have actual version for Chrome also. Lazarus for Chrome was broken, but someone fixed the version 3.0.5 to work on new versions of Chrome.
[lazarus-form-recovery_3.2.zip](https://github.com/lazarus-recovery/original_ff/files/1272633/lazarus-form-recovery_3.2.zip)

**Comment by sierracircle** [comment]
I have posted a job on upwork to find a qualified Firefox Add On Developer who can give us a quote. Once I have a reasonable quote, we can find out if we can raise a collective amount of dollars for that.

**Comment by sierracircle** [comment]
I have the project posted with a budget of $400. A couple of people have applied for the project, but I will wait for a while to see if anyone else applies.

 I figure all of the people who have offered to contribute can pitch in a little until we have that much. I will add more info once I discuss the project with some of the people applying.

**Comment by sierracircle** [comment]
Ok. I have had several people apply for the project so far, with fees between $350 and $400.
If you would like to donate towards this, you can send to my Paypal at my email address. I will put a notification up when we have enough funds>   adam@sierracircle.org

There are about 8 of us so far who have expressed interest/willingness to donate funds. If we each pitch in $50 it should cover the initial build, but donate what you are able. I will cover the remainder.

@nh2 
@guysoft 
@Tormen 
@chickendude 
@g3rman42 
@franzalex 
@sNiXx

**Comment by Tormen** [comment]
I am in with $25.

**Comment by chickendude** [comment]
I just tossed in $25.

**Comment by sierracircle** [comment]
..up to $75.00 so far.

**Comment by maphew** [comment]
I'm in for $30

**Comment by NoelLH** [comment]
Just pinged you $25.something

**Comment by nh2** [comment]
OK, $75 from me to get Lazarus Recovery back into shape!

I'll be happy to add another 75 once it has been confirmed that the WebExtensions implementation will store form texts encrypted behind the masthttps://www.upwork.com/job/Migrate-Firefox-xpi-Updated-Firefox-Addon_~015ad6184394b002b9/er password, like the original Lazarus did.

@sierracircle This is your upwork post, right? https://www.upwork.com/job/Migrate-Firefox-xpi-Updated-Firefox-Addon_~015ad6184394b002b9/

**Comment by captainhook** [comment]
I'll contribute $20.

Who would maintain the final product?

**Comment by ExE-Boss** [comment]
Does Seth Wagoner know about this, and is he fine with this?

### Seth’s online profiles:
https://twitter.com/sethop
https://www.facebook.com/sethop
https://www.linkedin.com/in/sethop
https://web.archive.org/web/20110716043232/http://sethop.com/ (old backup from back when the website was still online)

**Comment by mavhas** [comment]
Hi, is $400 a reasonable estimate (are you getting the right level of expertise responding)?
If you can confirm the budget is realistically set, I'll be happy to make a donation of $50 at this time.

**Comment by BubiBalboa** [comment]
I did some sleuthing and found a Goodreads account which has recent activity by the original dev Seth Wagoner.

So we could just ask him if he is okay with this project.

https://www.goodreads.com/user/show/6043163-seth-wagoner

Any takers?

**Comment by sierracircle** [comment]
We are currently up to $275.

@nh2   yes, that is the Project I have on Upwork. So far 4 people have applied for the project.

@captainhook   The project will be maintained as an open-source project here on Github. I currently have the lead, but if someone else desired to take over as lead I can turn that over.

@ExE-Boss  I have attempted to contact Seth on multiple addresses/platforms for the last two years, with no response. Judging from all the comments on the Add-on page, no one else has managed to get a response from him either.

@mavhas  The three different people who have applied for the project on upwork have given estimates between $350 ~ $400.  One person wants to be paid hourly, but I am not ready to get into a potentially never-ending pay-cycle.  Two of the people have resumes/work history of working with Firefox and Chrome add-ons. They both are studying the code at the moment and will get back to me when they have a better idea of what will be involved.

All of the people who have applied have said that the better approach would be to work with the Chrome Extension and convert that to Firefox rather than converting the Firefox xul to the updated format.

Have any of you used the Chrome add-on?  I am going to play with it now, if it even works. There are two versions in the Chrome web-store: Lazarus Form Saver, and Lazarus Form Saver - Patched.

**Comment by maphew** [comment]
@BubiBalboa Sure, go for it. Point him to this issue and the UpWork project. Maybe seeing this effort to reboot the project will re-awaken his enthusiasm for it. (How do you know it's the same Seth?)

**Comment by ExE-Boss** [comment]
@sierracircle Seth Wagoner currently works at [MetaGeek LLC](https://www.metageek.com/), have you tried contacting him through that?

**Comment by mavhas** [comment]
Sounds excellent, Adam.
Looking forward to any progress with great enthusiasm.

Hope there'll be more donations.
If push comes to shove I'll be fine to offer some more. The Lazarus extension has saved my life (well, that's a bit dramatic. :) Let's keep it at 'a lot of time and frustration') multiple times.

**Comment by BubiBalboa** [comment]
@maphew I can't be certain, of course, but the Goodreads profile has the same picture as his Twitter page. Chances are the profile is legit. If he wants to talk is a different story.

@ExE-Boss  I don't think he would appreciate being contacted through his employer.

**Comment by sierracircle** [comment]
@ExE-Boss   Honestly, I feel that I have made a reasonable effort to contact Seth. I am not going to pursue it any further.  Reading through the comments, it sounds like more than one person paid for the unlocked version and never received anything. I hope to rectify this by offering the unlocked version free to everyone

**Comment by maphew** [comment]
_Honestly, I feel that I have made a reasonable effort to contact Seth._

Agreed. I have made a few attempts also. Thinking on it more, pursuing him
through Good Reads is probably best left alone, and definitely not through
his employer. That changes the situation from 'contacting' to 'hounding'.

He made the project open source. You don't do that unless you intend it to
be used and picked up by others. He's had ample opportunity over the years
in several places to express his wishes for Lazarus and hasn't. It's fair
to conclude Seth is just not interested anymore.

**Comment by guysoft** [comment]
Sent $25, just got back now from Burning Man. 

Please keep us posted.

**Comment by sierracircle** [comment]
Thanks @guysoft  for the donation. we are now up to $300.  I have a check for $50 coming soon for a little project I just finished, so I will contribute that as well. 
The current applicant I am messaging has quoted $350. I am waiting to hear back from him concerning a couple of questions, but if he says he can do it I will hire him.

**Comment by Tormen** [comment]
@adam:
Thanks for all the feedback! Great to hear about the 2 people with FF
work history 

Given the similarity between the new Firefox API and Chrome extension
API the Chrome -> FF API (rather than Xul -> FF API) seems indeed best.

I am happy that there is some traction building up and things start
moving :)

I just put my money to where my mouth is and sent you $25.

**Comment by sierracircle** [comment]
Thanks @Tormen . We are at $325 now.

**Comment by sierracircle** [comment]
I have hired a person for the project. His name is Viktor. We have been messaging back and forth for the past week. He has experience with both Chrome Extensions and Firefox Add-ons, as well as building a form-filler.
He is starting with the Lazarus Chrome Extension, and will fix bugs there before converting it to a Firefox extension.
Once his work is done, We will need to send the updated version to Firefox to have it approved for the Add-ons website, and Google for the Chrome Web Store.

Does anyone want to take on that part of the project?

**Comment by adam** [comment]
@Tormen - I am not the Adam you're looking for. ;-)

**Comment by guysoft** [comment]
If he pushes to a public repo we might be able to have a look. But not a
requirement.

**Comment by sierracircle** [comment]
our guy is starting work. He is using the 'Lazarus Form Recovery - Patched' as the starting point. It is available in the Chrome Web Store.

He has requested we report any bugs we can find so he can fix those before porting it to Firefox.  It would be great if we could all test it thoroughly  and report any issues here:
https://github.com/lazarus-recovery/chrome_patched

I am closing this thread, as we are actively working on the new version, so this discussion has been resolved.

**Comment by Photon89** [comment]
Will there be more comments on the development progress here? If not, how can I stay informed? Thanks everybody for reviving Lazarus btw.!

**Comment by sierracircle** [comment]
@Photon89 Please see my comment just above yours.  This thread is closed and we will be reporting bugs on the chrome version here: https://github.com/lazarus-recovery/chrome_patched

Once we have the current version on Chrome worked out, it will be switched to FF and I will open another repository for that. Just keep an eye on https://github.com/lazarus-recovery

**Comment by Photon89** [comment]
@sierracircle Thanks!

**Comment by dartraiden** [comment]
Found [FormSave](https://addons.mozilla.org/en-US/firefox/addon/formsave/) extension. Not as functional as Lazarus, but it works.

**Comment by guysoft** [comment]
Is all good? No pushes to the repo since it was created 14 days ago.

**Comment by sierracircle** [comment]
I gave the project a month to complete on upwork. I will post nofifications and updates on a new thread.  I am locking this thread so people will stop posting to it.

**Comment by sierracircle** [comment]
So the first developer who I hired on UpWork  has said that the project was too complicated for him (specifically, he said that the IndexDB in Firefox is using up tons of memory). He has resigned from the project. Upwork is refunding me the money. I have re-posted the job, but everyone who has applied is bidding more than $400. (between $500 and $1500 so far)

We can either try to raise more money collectively, or I can just refund everyone who has contributed so far and we can call it quits.



What say ye group?

**Comment by guysoft** [comment]
Perhaps we should look for a different place to advertise / try increase thr budget? What is the budget now? I can ask on firefox IRC or somewhere people might have thr expertise

**Comment by jirutka** [comment]
I'm gonna send some money till the end of week and tell more people about it.

**Comment by maphew** [comment]
Kudos for the developer for resigning, and not just using up the time/money
and not leaving us and him/her hanging. That could have easily happened.
Please pass that on. Ask if there is anything in the project pre-work
investigation that would be useful for the next developer, and offer
recompense for at at least that much. (You can use my contribution)

I think we could raise an extra $100 for the next low bid.

Maybe there's a way to break it into smaller pieces? e.g. $200 for X?
(Though imagination fails me right now for what would be a smaller step
that's meaningful enough to pay for.)

**Comment by sierracircle** [comment]
@maphew   If I am able, I will pass that amount on to the developer, although I already closed out the contract, and upwork does not like me to do transactions outside of their website; however, I will send them a message and ask them if I can pay him something outside of the contract.

**Comment by sNiXx** [comment]
I agree with @guysoft that we should advertise this project elsewhere and @maphew that this was a very honest move by the developer. I would gladly increase my support if we can get more people on board.

**Comment by mavhas** [comment]
Happy to increase my contribution as well...An extra 50 OK?
Main thing is to get this thing scoped properly so it doesn't turn into a  bottomless pit though!  I can't do this indefinitely! :-)

**Comment by sierracircle** [comment]
@guysoft  currently we have $350 total. I was going to pitch in whatever the difference was needed when we hired someone on.  
I have been listing the project as $400, but when people bid they can bid for a higher or lower amount.

If someone wants to take on listing the project elsewhere, let me know and I will transfer the money I have to you and you can handle hiring and such.

Otherwise, I am sticking with Upwork, as I already have an account with them.

@mavhas  anything you are willing to pitch in will be great. I think if we get up to $500 or $600,  I have several people bidding within that range

**Comment by jirutka** [comment]
I sent $20 to adam@sierracircle.org.

**Comment by nh2** [comment]
> I'll be happy to add another 75 once it has been confirmed that the WebExtensions implementation will store form texts encrypted behind the master password, like the original Lazarus did.

OK, I've sent another $75.

**Comment by ronrubin** [comment]
Hi everyone,
Just noticed this thread randomly when browsing the Lazarus comments section. Would be glad to chip in. Too bad though that this is not getting more publicity - I'm sure there is a much larger audience that would be willing to contribute. Any ideas on that?

**Comment by hyiltiz** [comment]
BTW, would be useful it this remains Open, since Issues doesn't show this by default now that is is "Closed". People who are interested in this project will definitely check on the Issues to learn about what is going on, and this is part of what is going on right now.

**Comment by sierracircle** [comment]
@hyiltiz   I have re-opened the discussion. It was previously closed because we had reached enough funding to hire a developer, but since then that developer backed out and we are having to hire someone else. 

So far, we have a total of $440.07. Here is a list of who has donated (I have used only first names). Niklas has generously donated twice a total of $150.00

- Adam		$50.00
- Michael      	$25.00
- Andreas     	$50.00
- Noel  		$25.15
- Niklas   	        $75.00
- Martijn		$50.00
- Guy		 $25.00
- Knuth		$25.00
- MATT		$27.77
- Pedro		$12.15
- Niklas		$75.00

**Comment by jirutka** [comment]
@sierracircle Why I’m missing in the list?

**Comment by sierracircle** [comment]
@jirutka  I do not have any incoming payments in my paypal from you. When did you send it? Does your statement say that it was sent? How much did you send? Did you use someone else's paypal account?

**Comment by jirutka** [comment]
@sierracircle I don’t see the payment in my history, so probably some error. I sent it again, $20.

**Comment by sierracircle** [comment]
@jirutka   that worked. It just popped up in my Paypal account. I added your name to the list.

**Comment by sierracircle** [comment]
Here is a link to the profile of a couple of developers I am talking to. The first is bidding $500, the second is bidding $1140:

https://www.upwork.com/freelancers/~017b984aebfbc54348

https://www.upwork.com/fl/peterv29

**Comment by gr3gw** [comment]
I've just chipped in $20.

**Comment by sierracircle** [comment]
thanks @gr3gw  I have updated the list above with donors. 

We have a total of $480.07, so I could potentially hire [Stanislav](https://www.upwork.com/freelancers/%7E017b984aebfbc54348) for the job, as he has bid $500. (I will pay the extra money)

**Comment by sierracircle** [comment]
Last chance for any input on the developer I am hiring, or if you want to move the project to a different venue for hiring that needs to done.
Once I hire a developer, Upwork takes the money for the project and holds it in escrow until the project is complete (or the developer backs out).

**Comment by hyiltiz** [comment]
He seems to have been successful in developing several simple add-ons, and based on that, it seems a good candidate. Did he provide his assessment about this job? I think we should know about what he thinks about this job before actually hiring him.

**Comment by guysoft** [comment]
Posted in the Mozilla forum: https://discourse.mozilla.org/t/looking-for-developer-to-hire-to-port-lazarus/20354

**Comment by sierracircle** [comment]
@guysoft   Do you want to take over the hiring process? I need to know soon if you do because I have a developer waiting to be hired on

@hyiltiz  I have asked Stanislav if he thought the project was within his skill level, and also told him about the issues the past developer had with the memory after using IndexDB.  He responded that he was capable of completing the project and DB issue would not be a problem for him.

 I like this developer for the project because he has a work history going back to 2012. Usually those developers work hard to keep their reputation good.

**Comment by guysoft** [comment]
@sierracircle I understood you were not finding developers who are capable, so I thought i might point some here. I don't want to take over the process. Also some users were also interested. Thought its a good place to make our cause known.

**Comment by sierracircle** [comment]
@guysoft   Got it.

**Comment by sierracircle** [comment]
Stanislav has accepted the project at $500.

**Comment by guysoft** [comment]
Yay! any news since the acceptance?

**Comment by KaiKemmann** [comment]
I send $20.
Also keep it for your previous effort if the project fails.

good luck to us
Kai

**Comment by sierracircle** [comment]
Thanks @KaiKemmann   I added you to the list.

@guysoft  I briefly heard back from the developer with something to show me, but whatever it was it was not working so he went back to the drawing board. I gave him over a month to work on it, so I am not rushing it.

**Comment by sierracircle** [comment]
Something to be thinking about/brain-storming:  currently the add-on has internal links to emails and website that are no longer active.
Should we create a new email/website for this project, or just direct people here to the gitbub page?

**Comment by maphew** [comment]
Perhaps https://github.com/barryclark/jekyll-now (haven't used it, but it looks low maintenance)

**Comment by Rotozaza** [comment]
I haven't read entire thread, did someone pick-up work as specified on Upwork? 

Budget need? 
I won't leave FF, but desperately need Lazarus.

**Comment by sierracircle** [comment]
@Rotozaza   We raised $500, and hired someone on Upwork. They have been working on it for the last month. I have not heard anything for the a week or so. He seems to be stuck at the same point the last developer was stuck at: converting the database storage from SQlite to something the newest FF can use.

**Comment by guysoft** [comment]
Can you ask him to post the problem? I spoke to FF developers about this. We could solve ot for him so he can focus on the rest.

**Comment by sierracircle** [comment]
@guysoft  I have sent a message to him just know asking about that. I will post back when I hear from him.

**Comment by sierracircle** [comment]
I received a reply from the developer this morning:

> Stanislav Golovin
> 3:35 AM PST, 9 Nov 2017
> Hi Adam
> 
> Yes, it's a bit slow, but I going forward. IndexedDB doesn't support sql, so it's a lot of work change DB layer in this addon. I hope I can finish later on this week (new version can be used in chrome too).

**Comment by guysoft** [comment]
There are libraries that convert it to SQL. For example:
https://github.com/agershun/alasql/wiki/IndexedDB
AlaSQL.js - JavaScript SQL database for browser.Handles both traditional relational tables and nested JSON data (NoSQL). Export, store, and import data from localStorage, **IndexedDB**, or Excel

**Comment by guysoft** [comment]
There are also more [talks about this here](https://stackoverflow.com/questions/6116053/javascript-library-to-bridge-indexeddb-and-websql).
[This was recommended there](https://github.com/treojs/treo)

**Comment by sierracircle** [comment]
It sounds like he is making progress and plans to be finished soon. I let him know there was help offered if he needed it.

**Comment by NormG1** [comment]
Just sent $20 to adam@sierracircle.org

**Comment by NormG1** [comment]
I can hardly wait until this is finished so I can update FF beyond V52.

**Comment by sierracircle** [comment]
Thanks @NormG1   I received the payment and updated the list above to show your contribution.

I am also excited about getting Lazarus working in the most recent FF. I have been working without it for the last few months and frequently wish it was there after having to fill out a tedious form more than once.

**Comment by Spidersouris** [comment]
Thanks for your work. Any update? I desperately need this.

**Comment by sierracircle** [comment]
@Spidersouris  no word yet from the developer. I set the time-limit for the end of November, so he has until then to show his work.

**Comment by OrangutanGaming** [comment]
End of November \o/

**Comment by sierracircle** [comment]
Still no reply. I have sent him a message just now and will report back after I hear from him.

**Comment by sierracircle** [comment]
If he requests more time we can either give him more time or find a new developer. The contract is up  November 30th.

**Comment by nh2** [comment]
A user [on reddit](https://www.reddit.com/r/firefox/comments/6yezci/crowdfunding_to_bring_back_lazarus_recovery_with/dqkrjkl/) pointed out the addon [Form History Control](https://addons.mozilla.org/en-US/firefox/addon/form-history-control/) just now that seems to have similar features to Lazarus.

I don't know yet if it covers all use cases of Lazarus and whether the contents get encrypted with the master password like in Lazarus, but it may be a useful thing to look at in our effort.

**Comment by maphew** [comment]
I would ask for whatever work has been done to date. Working or not, there
are good things to learn from false paths and broken ideas. Sometimes
knowing what not to do or what can't work is most important.

It does feel a little strange for work on an open source project to be done
out of sight. I can understand not wanting to have dozens of eyes of paying
customers watching your every move though.

**Comment by colans** [comment]
> A user on reddit pointed out the addon Form History Control just now that seems to have similar features to Lazarus.

Yes, after doing a bit of research it seems like a reasonable alternative.  I started using it, and it works, although differently.  The code is available at https://github.com/stephanmahieu/formhistorycontrol-2.

**Comment by Tormen** [comment]
The developer behind History Control 2 is really motivated.

See his recent (October 2017) comments:

Hi, webextensions can no longer access the browsers formhistory, so
unwanted history can no longer be removed by this or any other
extension. The only way is indeed by erasing the complete formhistory,
or if you manage to get it displayed in the built-in auto-complete
dropdown list, you can remove an unwanted item by pressing the delete
button.
My new extension is in fact storing it's own copy of everything you type
into textfields. I am currently investigating the possibility of adding
my own autocomplete-dropdown list which will hopefully do a better job
as well.

OR this one from 12th of November:

Yes, the possibility to exclude certain sites and/or certain fields ranks high on my todo list.

See here:
https://formhistory.blogspot.de/2017/03/form-history-control-ii.html

I suppose, it would really be nice to see how to converge our effort and
his effort.

Maybe pay him?

...

**Comment by sierracircle** [comment]
I received a message from the developer this morning:

> Yes, I'm working on it. I found better solution in last week and did some rework(
> Hope can send it to you in 2-3 days

**Comment by guysoft** [comment]
so..... today?

**Comment by sierracircle** [comment]
I have not heard anything back yet. I will drop a line later this 
evening to check the progress.

**Comment by NormG1** [comment]
What is the plan if the developer fails to deliver the product as promised?

**Comment by sierracircle** [comment]
The last time the developer could not complete the task, the money was refunded to my Paypal account (it is held in escrow by Upwork while the project is ongoing). 
I assume they will refund my money again, and I could then refund each person who has donated and we can come up with a different plan of action.
Alternatively, I could hold on to the money and we could try to find another developer.

However, this is all speculative, as the current developer last said they had made progress, so I am just waiting to see how this pans out.

It would be nice to have it functional before the end of they year.

**Comment by NormG1** [comment]
Have you seen any work product? Even if it isn't ready for prime time, it would be nice to know how far along they are and what problems are coming up.

If they fail, I'd vote you keep the money and look into finding a new developer.

**Comment by sierracircle** [comment]
I left a message just now asking him to show his current progress. If he can not show anything, I will start closing the project with him and look for someone else.

**Comment by sierracircle** [comment]
the most recent reply from the developer. I will give him until Monday:

> It's still same - migrate from sql to plain indexDB... As I say I found google lib which support sql syntax for any DB engines. And I'm working to integrate it. On this week I take a small break - first version almost done anyway. I think I'll send it to you no later then Monday, maybe earlier but I'm not sure - I have to check all db queries in addon.
> 
> I'm sorry for that delay, I thought that will be much easier and faster.

**Comment by sierracircle** [comment]
I received this message this morning:

> 
> Hi Adam
> 
> It works with indexedDB
> 
> I'm removing unused old code\sync etc.. 

I take it he is wrapping things up.

**Comment by NormG1** [comment]
Exciting news. Quite a few web sites are no longer backwards compatible with V52.

I will be so happy to be able to upgrade to V57.

Does it need to go through Firefox approval to be an official add-on or will we be able to load it directly?

Will the developer respond to any bug reports that might arise? And will he expect payment?

Who actually owns the code?

Norm Gilbert   Tour Guide, Grayline SF	
	
Phone: 415-465-4220
Email: fogcitynative@gmail.com
Skype: fogcitynative
Website: http://www.graylineofsanfrancisco.com
Address: 355 Serrano Dr. #12A, San Francisco, CA 94132

"The idea for the work party grew out of offers of help from alumni and retired faculty who gathered at memorial services held for crash victims." - G...

**Comment by sierracircle** [comment]
**- I will be so happy to be able to upgrade to V57.** 
      Me too!

**- Does it need to go through Firefox approval to be an official add-on or will we be able to load it directly?**
     I do not know yet. That will be the next thing to tackle (contacting Mozilla and finding out what needs to be done)


**- Will the developer respond to any bug reports that might arise?** 
       You never know with developers. Some do, some do not respond to bug-fix requests. We will find out soon enough I imagine

**- And will he expect payment?** 
       do not know, see above

**- Who actually owns the code?**
       As far as I am concerned, it is open-source, so everyone!

**Comment by OrangutanGaming** [comment]
So he's still "wrapping up"? Or is it done and pending to be posted on the addons page? We should still be able to import the file if adding it to the store is going to take time

**Comment by sierracircle** [comment]
Sorry for the delay. My kid was in a Christmas Pageant last night, so I was busy all day. Here is what the developer posted:

> Here is new version. It works only in Firefox.
> 
> To check it in firefox open this about:debugging#addons - click Load temporary addon and choose manifest.json
> 
> If you plan to support chrome - would be better to update chrome to this solution also. Migration from old DB can be added (I mean extension can read saved data and write it to new db).
> Also as I see used chrome api (for extensions) outdated - I changed all deprecated api calls. But extension still uses very old jquery. Refactoring also will be good.
> 
> Really sorry for this delay (stuck with indexed DB)


and here is the zipped extension:
[https://sierracircle.org/files/WebExtension.zip]WebExtension.zip(url)

I just tested it on a form and it was working. Please have a look and let me know if you have any issues so I can tell the developer.
If I do not hear back in a day, I will close the project so the dev can get his payment.

**Comment by dartraiden** [comment]
> As far as I am concerned, it is open-source, so everyone!
It's bit complicated. "Open-source" is not "free". So, if old Lazarus was not "free" - all rights belong to the author of the old Lazarus, because new is derivative work.

**Comment by guysoft** [comment]
@dartraiden No, if its a rewrite with none of the original code, then its a completely separate legal entity. You are mixing between patents (who could limit the invention) and code release.

I warmly suggest you release now with GPL v3 ASAP, thats what I backed for. Don't just say "free" without an explicit license.

**Comment by dartraiden** [comment]
Hmmm... I see [this](https://i.imgur.com/N7Mf2Pi.png) (pointing to getlazarus.com)
and also strings like
`URL_UPDATE_CHECK: 'http://getlazarus.com/updates/update-check.php?platform={platform}&version={version}&channel={updateChannel}&format=json',`
`One day I'll write a FAQ entry that explains this a bit more, but for now you can email me (lazarus@interclue.com) if you've got any questions about this`
that's why I think about code reusing.

OK, if extension completely written from scratch - no problems.

**Comment by guysoft** [comment]
We paid a developer to rewrite it :)
@sierracircle any installation instructions to this zip? Have you tried installing it yourself?

**Comment by guysoft** [comment]
Ok, seems to load, there is really some code-copy I suggest we remove so this would be a rewrite and we could release it open source. 

Some images from the extensions
![screenshot_20171217_025221](https://user-images.githubusercontent.com/325670/34075672-da5799b0-e2d5-11e7-8b45-6dce04dd6fc7.png)

Can't wait to push code to change this:

![screenshot_20171217_025155](https://user-images.githubusercontent.com/325670/34075673-dab56432-e2d5-11e7-9d77-7a862eeb09b0.png)

**Comment by sierracircle** [comment]
@guysoft 

from my previous post: 
To check it in firefox open this about:debugging#addons - click Load temporary addon and choose manifest.json

in simpler terms:

- Type this in the address bar    about:debugging#addons
- click Load temporary addon 
- choose manifest.json  from the unzipped folder you just downloaded

**Comment by jirutka** [comment]
That's great news!

However, I'm quite  concerned about this delivery. A zip file, really?! I hope he will eventually publish his work into a git repository, with full history, so we can reasonably review it in context with the original code.
I'd personally never accept just a zip file with bunch of code.

About the license, I'd personally prefer MIT, BSD, ISC or Apache license (well, basically any OSI license except GPL).

**Comment by DarkVoyage** [comment]
Should have a new icon too, that's also a piece of old plugin that can't be used without attribution.

**Comment by sierracircle** [comment]
Listen people..we will tackle all those things. one step at a time.

For now, can you please just test it and report if it is working or not?

**Comment by DarkVoyage** [comment]
It works. Thanks for all your hard work.

**Comment by sierracircle** [comment]
will someone test encrypting with password and let me know if that works?

**Comment by guysoft** [comment]
@sierracircle we can't handle all issues on this issue. We should open a git repo with the code and open individual issues there.

**Comment by sierracircle** [comment]
@guysoft   you are right, and I am going to do that, but at this moment in time I need to release the project on Upwork so they will pay the developer for the work he has completed.

If the basics are working he should get his money.

We are leaving in a couple of days for Christmas break and I would like to not spend my break doing that.

**Comment by maphew** [comment]
I can take care of putting the code into the repo. Enjoy your holidays and
family Adam, and thanks for shepherding the project over the last while. :)

**Comment by sierracircle** [comment]
Hi @maphew    I appreciate it. however, at the moment what I really need is not for someone to put the code in the repo, but for people to test the add-on and tell me if they have any glaring issues.

I want to pay the guy before I leave. I am waiting to hear back from everyone. So far only one person has actually responded that they have tried it and it is working.

CHECK THE BLOODY ADD_ON!!! IS IT WORKING FOR YOU? RESPOND YES OR NO!!!!!  THE DEVELOPER IS WAITING FOR ME TO RELEASE THE PROJECT.

once I hear back from a few people, I will release the project on UpWork and we can add the code to a repo.
we can deal with everything else in the coming month. 

Does that make sense?

**Comment by iiiGerardoiii** [comment]
I'm testing it, works good so far 👍

**Comment by Spidersouris** [comment]
Doesn't seem to work on Nightly 59.0a1 (2017-12-15) (64-bit) unfortunately. Clicking on the Lazarus icon at top right doesn't do anything, and there is no button near forms (right-clicking doesn't show any Lazarus-related option either).

**Comment by sierracircle** [comment]
here is a signed link if you want to install it beyond the de-bugging method (which removes it after restarting firefox):

https://sierracircle.org/files/lazarus_form_recovery-3.2.2-an+fx.xpi

**Comment by sierracircle** [comment]
@Spidersouris  can you access the settings from Add ons>Lazarus>Preferences?

I am able to make a password, but it does not let me put in a password to unlock the saved forms (nothing happens)...anyone else have this issue?

**Comment by OrangutanGaming** [comment]
Minor bug: When you increase the time to 14 days on the page, try to go above that, close the popup, close the options page, then reopen it, it says 15 days

**Comment by OrangutanGaming** [comment]
Also, with a password, when I try to click the button and click enter password, it does nothing

**Comment by Spidersouris** [comment]
@sierracircle I can, yes, but the password doesn't even seem to be saved. Also, the forms are still the same, there is no Lazarus icon or whatever.

**Comment by Photon89** [comment]
I didn't participate in the fundraising (a poor student ;)) so I don't feel entitled to pose feature requests but just in case it bugs anybody else: The context menu (what appears when clicking on the Lazarus icon next to an active text field) is not following the desktop style guidelines. It looks like it would fit in nicely into a recent Windows desktop but on other platforms (older Windows versions, Mac and Linux) it doesn't.

**Comment by sierracircle** [comment]
I had to install the add-on from file rather than de-bugging to get the password to be able to enter...still does not work though.

**Comment by OrangutanGaming** [comment]
I installed it from the link

**Comment by Photon89** [comment]
Firefox has just been updated from 57.0.1 to 57.0.2 and all the saved forms from before the update seem to be gone (in particular, my previous comment in this report has been saved by Lazarus first but is not available after the update).

**Comment by chickendude** [comment]
I've just tested it out a bit and apart from the options page with the donation/website/support links that appear to do nothing anymore, it works for my needs. I never used the password function before, i just tried adding it but now there's no actual option to input the password to unlock the saved form data. I won't be able to fully comment for another week or so after i've had a few lost post situations, but so far it seems to work just fine (minus the password, which i doubt i'll use anyway).

**Comment by Photon89** [comment]
I might be wrong in the interpretation of my observations. Now also my second comment is not available any more with no updates inbetween. Do you guys have all your old Github comments listed by Lazarus on the corresponding text fields?

**Comment by chickendude** [comment]
@Photon89 what do you mean? I just checked and only the last version of what i wrote is available. Perhaps it stays there until you switch pages?

Also, not sure if the old Lazarus did this or not, but it doesn't show up for contenteditable="true" objects, for example the post box at the bottom on this page is a `<body dir="LTR" style="overflow-y: hidden; min-height: 99px;" contenteditable="true"></body>` element and doesn't save form data:
https://forum.wordreference.com/threads/euskera-irrifarra.1219918/#post-17258659

**Comment by Photon89** [comment]
For me, in this github form I am currently writing into no version is available. In old Lazarus btw. all the former inputs were saved and shown in a list.

**Comment by sierracircle** [comment]
If you use the de-bugging method to install the add-on, it will not save between Firefox sessions.
If you install it from the xpi I created, it will save forms.

I can see my last input in this github discussion. I am using FF nightly on Ubuntu.
It is also updating as I type new things..so works for me

**Comment by Photon89** [comment]
Unfortunately, I used the xpi (thanks for it btw.)...

**Comment by sierracircle** [comment]
@Photon89   Does it save forms between restarts of firefox?

**Comment by Photon89** [comment]
@sierracircle : No, that's exactly my problem.

**Comment by NormG1** [comment]
V 57.0.2 Mac High Sierra Not working. Turned on password preferences. Cannot enter the password on the pop-up when clicking the Lazarus icon. Lazarus options will also not pop-up. Trying restart.

**Comment by sierracircle** [comment]
@NormG1    does it work if you do not use Password Protected?

**Comment by NormG1** [comment]
Nope. It is simply not saving form data at least from this site and these forms. I haven't tried any other sites but old Lazarus would have saved all keystrokes in any form field. And this is a form field.

**Comment by NormG1** [comment]
Nope. It is simply not saving form data at least from this site and these forms. I haven't tried any other sites but old Lazarus would have saved all keystrokes in any form field. And this is a form field.

**Comment by NormG1** [comment]
Also, I miss the old database box so I could b ring up EVERYTHING Lazarus has saved and now the only entry point is via the pop-up in a specific field. 

Removed the add-on on the possibility that in overwriting the old one something burped. Re-installed. The add-on is now in the list of add-ons and I can access the preferences through the add-on listing. But now I have no Lazarus icon in the upper right of text boxes.

**Comment by NormG1** [comment]
Got the Lazarus Icon on a Firefox restart. There is saved text. But it is only present while I am entering text. If I delete the text, say by accident, it also deletes the text in the Lazarus database. So selected ALL after I have typed this and pressing delete leaves nothing in the database to be able to retrieve. So maybe it is useful if a website times out and then I press submit and lose my work but need to test that on a website I know has a short timeout. But sure does not work anywhere near as well a (HERE IS WHERE NEW LAZARUS STOPPED SAVING TEXT. IT APPARENTLY HAS A CHARACTER LIMIT ON WHAT IT WILL SAVE. Here's the rest of what I wrote, none of it saved by new Lazarus)

s the old Lazarus did. Old Lazarus, if I selected all and deleted, would have saved everything up to that point like a keystroke recorder and closed the entry. If I started typing something new, it created a new entry in the database. So as I filled out a single form there were many entries in the Lazarus database and I usually would be able to find the latest since they were all time stamped.

**Comment by NormG1** [comment]
Once I clicked the COMMENT button a new text box opened. Before I started typing this, the old entry up to the character limit was present in the pop-up. God knows how I would transfer it to the clipboard and paste it into the new box. And once I started typing in the new box, it started saving by overwriting the previous saved entry. It sure isn't saving for 14 days. Not sure if the database is indexing by website, such that my typed comment here will be preserved should I come back here and be presented with an empty text box. But going to different website opens separate entries in the database.

**Comment by NormG1** [comment]
OK, figured out how to get a Lazarus entry into a new text box. It doesn't use the clipboard or cut and paste. It just populates the empty box automatically. But hit one keystroke and what you typed is lost forever.

**Comment by NormG1** [comment]
Off to test timeouts.

**Comment by NormG1** [comment]
The icon in the menu bar does nothing when clicked on.

**Comment by NormG1** [comment]
So far it is a HUGE disappointment when compared to Lazarus. It barely protects a user that spends a lot of time composing some text, presses submit and hits a timeout. It doesn't even save the whole thing because it is limited to a certain number of characters.

If what was asked to be done in the spec was to make Lazarus work with newer versions of Firefox, I don't think this developer had a clue. He sure didn't update Lazarus. He made some little tiny applet to do only one thing and yes, it does that one (NEW LAZARUS HAS STOPPED RECORDING KEYSTROKES HERE) thing but not very well.

Did the developer even get an old version of Firefox and run the old Lazarus to get a feel for what it did and how it did it? Did he ever have screenshots provided to him?

On a 5 star scale, this is a 2-star applet.

**Comment by Photon89** [comment]
From skimming the above discussion, I got the impression that a complete rewrite is what the dev actually had to do, since not much of the original code was usable with FF 57.

**Comment by NormG1** [comment]
It sure doesn't seem like a month's worth of development and coding.

**Comment by NormG1** [comment]
I also can't get the icon out of the FF menu bar, since it serves no useful purpose after the first time.

**Comment by Photon89** [comment]
It should actually show if Lazarus is applicable for the currently open website.

**Comment by NormG1** [comment]
No to saves over time-outs. I was on my medical insurance web site. Started typing a message to my doctor. Left it alone without trying to send. Waited for the time out. Went back into message composition and of course, the new Lazarus database was empty and what I had typed so far was lost.

Unless the message is actually submitted but fails, perhaps then something is saved for a while. But without a key click on a send button, nothing is saved into the new Lazarus database.

Sorry but this app is shit . . Pretty useless. I wonder what was in the original spec posted on the site to which this developer bid the job? 

It almost reminds me of the kind of work you get from developers in India. They are technically competent at coding but if you do not spell out exactly what you want right down to screenshots and page designs, you get what they think you want, which doesn't actually show much logical thought into how something works in the real world.

So this thing likely does what the developer intended but it is just useless in actual function. 

Actual function might be "I want a keystroke recorder operating in every web form that is constantly working in the background to save every keystroke as it is typed and to save all those keystrokes in an easily accessible date and time stamped non-destructive database from which the text can be copied and pasted into any other web form forever or until the user deletes one or all of the entries in the database. There is no limit on the amount of text that is captured or the number of open web forms. If it fits into the text box of the web form, it gets captured. Saving is done constantly from a cache but there is no visible slow down as this app works in the background. It must be password protected and survive across crashes, timeouts, restarts, or any other typical hazard that causes a user to lose their compositions. Compositions from different websites are saved separately. The database is periodically backed up to the cloud.  And it must be customizable enough to not capture passwords and to be able to be disabled on specific websites." 

I welcome any edits to improve this description of the function of Lazarus as I see it.

**Comment by Photon89** [comment]
What kind of clouds are you thinking about? Did old Lazarus had this feature? I'd prefer to have it stay on the local disk actually.

**Comment by sierracircle** [comment]
Ok my fine friends. The developer has heard our cries and did an update in short time. Try this and report back:
https://sierracircle.org/files/lazarus_form_recovery-3.2.2.689-an+fx.xpi

**Comment by sierracircle** [comment]
I am having better luck with this newer update:
Character limit seems to be fixed
password is working now
saves at least two versions of form

**Comment by sierracircle** [comment]
..still losing forms after a certain amount of time...reporting to dev.

**Comment by maphew** [comment]
I've created a branch of the Up-Work developer's code, created a tagged release for the signed xpi @sierracircle  posted earlier, and started a wiki page for how to test. Please feel free to flesh it out and correct as needed.

* Release: https://github.com/lazarus-recovery/original_ff/releases/tag/3.2.2-an%2Bfx
* Code: https://github.com/lazarus-recovery/original_ff/tree/web-ext-2017-12-16
* Wiki page: https://github.com/lazarus-recovery/original_ff/wiki/Testing-development-version

It's probably best if feedback for what works and what doesn't are posted as new issues. It's very easy to lose information in this long thread.

**Comment by sierracircle** [comment]
Ironically, this github discussion page seems to be the culprit in throwing things off. If I do not visit this page at all, other forms are saved just fine.
Coming to this page or leaving it open seems to throw the other saved forms out...I am going to test this theory.

**Comment by maphew** [comment]
***[Release](https://github.com/lazarus-recovery/original_ff/releases) updated to v3.2.2.689.***

Works on Linux Firefox 57.01 (64bit), though there's a lot of testing still to do.

**Comment by OrangutanGaming** [comment]
Will it be added to the addons page when the debugging is done?

**Comment by Photon89** [comment]
With the xpi linked in the release by @maphew not even the last form is saved within a session, as soon as I delete the contents of a form, the list of forms gets empty. (Always testing in the github page btw.).

**Comment by sierracircle** [comment]
Okay, third iteration. Test and report back:
https://sierracircle.org/files/lazarus_form_recovery-3.2.2.690-an+fx.xpi

**Comment by Photon89** [comment]
Cool, now it starts to function! I wrote some text in this form (without submitting it), it was shown in the Lazarus list and still is, after Firefox restart.

One less dramatic thing (probably a feature and not a bug, but this is debatable): I wrote one line, it has been saved to Lazarus. Then I added a second line and it created a second entry (first entry including the shorter text, second entry the longer one). This is so far the desired behaviour. However, I then deleted the second line again and the second entry in Lazarus has been deleted as well as if I never typed it. Is this to be considered a bug or a feature? I think, it would make more sense to save text even if it has been deleted by the user, at least this is how the old Lazarus used to work, as far as I remember.

**Comment by Tormen** [comment]
* Logo is still the original one? This should change!
* The "buy pro version" link should go away (on settings page)
* The windows XP style icons for the popups (e.g. when setting a password) are not nice :(
* Data expiry of 14 days is rather bad. I always had setup 999 days in Lazarus. There are plenty of forms that I need to access only once a year!
* There is currently a "An error occured when checking for updates." error message on the settings page

**Comment by sierracircle** [comment]
@Tormen   All of the things you mentioned are great issues for us to work on once we actually put this up on github. I hope you will open an issue for each one once posted.

At the moment, we are just looking at core functionality so I can pay the developer and we can move on to the next phase of things (having the new plugin in github and open to issues, contacting Mozilla about updating the old Add-on page, etc)

**Comment by sierracircle** [comment]
@Photon89   it is also essentially working for me as well, except now "Disable Lazarus for this site" has stopped. I am reporting this to the developer.

**Comment by OrangutanGaming** [comment]
Ok, most of the bugs beforehand are fixed, don't seem to see any issues, but when you have some text and you accidently select all and delete it, it doesn't save which is kind of annoying, especially if you can't undo it

**Comment by sierracircle** [comment]
Here is one more, where he fixed an issue with "Disable Lazarus" was not working for some sites:

https://sierracircle.org/files/lazarus_form_recovery-3.2.2.692-an+fx.xpi

The core functionality is working. In the new year we can tackle contacting Mozilla to get the old add-on page updated, working out the minor glitches, and getting the repositories sorted.

**Comment by sierracircle** [comment]
I have paid the developer so he can move on to other projects. He said he is willing to continue working on the project in the future and has some ideas for updates, migrating to Edge and Chrome and cleaning up the API calls.

**Comment by OrangutanGaming** [comment]
Getting it on the mozilla page will be super useful as we can properly update instead of removing the old version every time, or you can have like 10 of the same plugin

**Comment by OrangutanGaming** [comment]
Are all changes pushed to `web-ext-2017-12-16` now? Or is the latest just the XPIs that you post here?

**Comment by maphew** [comment]
@OrangutanGaming Seeing that we now have a usable extension all changes have been pushed to `master` and I've deleted the intermediate `web-ext-*` branch.

**Comment by maphew** [comment]
I created a [Publish to Mozilla Add-ons](https://github.com/lazarus-recovery/original_ff/milestone/1), with a tentative target date for end of January. Please use that label for any issues that must or should be resolved before then. To my mind those are (a) a security audit and (b) finding out if we can publish as an update to the legacy Seth Wagoner one or need to publish as new

**Comment by BubiBalboa** [comment]
Seeing that Seth Wagoner did not respond to any messages I think it would definitely need to be published under a new name (Lazarus Reborn/Phoenix/Something Form Recovery?) and some person has to own the new AMO page.

The addon should look very different from the original Lazarus to avoid confusion. New name, new icon, new look. Seth Wagoner could and should still be credited in some form, though.

Just my two cents as a lurker of this project.

**Comment by maphew** [comment]
A general note to everyone: I created this GitHub organization, have been helping @sierracircle with some administration tasks, and am noisy with the Issues and other organization type work. However I'm not the boss or leader of the project. I may say things that sound like directions, _"use this label for that problem"_ and _"we need this by that date"_, but please interpret them as suggestions. I'm just a dude picking up some things here and there that look like they need doing, because I want Lazarus to remain active and viable.

**Comment by maphew** [comment]
@BubiBalboa see #10

**Comment by OrangutanGaming** [comment]
I think contacting Mozilla to get ownership of it could be possible

**Comment by OrangutanGaming** [comment]
What are you actually using to build the XPI? I tried `7z a -tzip -mx9 -mm=Deflate -mfb=258 -mmt=8 -mpass=15 -mtc=on ^
   ..\dist\lazarus-recovery-dev.xpi * -r` from the batch file, but Firefox is saying it's corrupted, yet https://sierracircle.org/files/lazarus_form_recovery-3.2.2.692-an+fx.xpi works fine

**Comment by sierracircle** [comment]
@OrangutanGaming 
To create an XPI, mozilla wants it to be signed, so you have to create an Mozilla account, upload the zipped files in proper format, and then run it through their validator. It then spits out and XPI that you can install.

I think for consistency, and since we will be contacting Mozilla about taking over the Lazarus add-on page, we should have one person who handles that account, creating XPIs and contacting Mozilla about the old Lazarus add-on page. 

Does anyone want to volunteer for that? You should have good writing and communication skills since you will be interacting with Mozilla Foundation.

**Comment by OrangutanGaming** [comment]
Well I was planning on creating a script that for now, would pull from git, then make an XPI and install it for me until it's added to the store

**Comment by sierracircle** [comment]
if you were just de-bugging then it is not neccessary to have an xpi.  To check it in firefox open this about:debugging#addons - click Load temporary addon and choose manifest.json from the un-zipped files.

**Comment by OrangutanGaming** [comment]
Isn't it then deleted after restarting?

**Comment by sierracircle** [comment]
@OrangutanGaming   correct. it is removed from FF after re-starting, but in some ways that is what you want if you are working on it directly, as it allows you to make changes to your files while working..then you can reload it without restarting:
https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Temporary_Installation_in_Firefox

**Comment by maphew** [comment]
Well I was planning on creating a script that for now, would pull from git,
then make an XPI and install it for me until it's added to the store

That was my plan too. The existing build script in the code repo is from
before FF changed to only allowing signed XPI's. There used to be an
`about:config` preference to turn that off. Now I believe you have to
running Firefox Nightly or Developer edition in order to use unsigned XPI's.

**Comment by maphew** [comment]
... we should have one person who handles that account, creating XPIs and
contacting Mozilla about the old Lazarus add-on page.

+1

**Comment by OrangutanGaming** [comment]
When this person is decided, I can use their API stuff to have it automatically update the plugin on the page from the current GitHub build

**Comment by mausalimi** [comment]
https://sierracircle.org/files/lazarus_form_recovery-3.2.2.692-an+fx.xpi

the domain doesn't seem to be working.

**Comment by maphew** [comment]
https://sierracircle.org/files/lazarus_form_recovery-3.2.2.692-an+fx.xpi

the domain doesn't seem to be working.

You can get the xpi from here also:
https://github.com/lazarus-recovery/original_ff/releases

**Comment by mausalimi** [comment]
thanks i even knew this, but it wasn't there when i checked first time (must have been 8 days ago, just between the posting from  sierracircle above and the upload to github... today i came back and only checked the sierracircle link. my fault.

so yes, 3.2.2.692 is available.

**Comment by sierracircle** [comment]
..sorry about the sierracircle.org link not working. My DNS server was having issues while I was out of town. It is fixed now.

**Comment by sierracircle** [comment]
>  we should have one person who handles that account, creating XPIs and contacting Mozilla about the old Lazarus add-on page.

Do we have any volunteers for the position?

**Comment by OrangutanGaming** [comment]
I guess I don't mind. I did try to find a way to contact them on their site a while ago but couldn't find *anything*. Only thing is a public forum, which I don't think is the place for this.

**Comment by sierracircle** [comment]
Actually, it looks like they have re-vamped the Add-ons section, and it does NOT include Lazarus in the listings as far as I could see.

I think this will make things easier, as you can probably now just start an account and add the new add-on.

It would be good if you shared the login information of that account with me and @maphew  so we don't get locked out if something ever happens to you.


So for now, just take the approach that this is a new add-on. Create an account and upload it to the Mozilla Add-ons and we can start updating it from there.

**Comment by OrangutanGaming** [comment]
Seems like there's an issue with their site and posting add-ons. I'll try again tomorrow as soon as I wake up. I also intend on having a Discord for it so support can be easier, and to talk to other devs for e.g. distributing logins

**Comment by OrangutanGaming** [comment]
I have a lot of experience with Discord, and think it's the perfect platform right now for something like this

**Comment by sierracircle** [comment]
any luck @OrangutanGaming  with the Mozilla Account?

**Comment by OrangutanGaming** [comment]
![image](https://user-images.githubusercontent.com/18350092/34424525-33f25ccc-ec1c-11e7-9fa0-4ede3de6345f.png)

**Comment by sierracircle** [comment]
Ok. We can wait for a while longer before uploading. Honestly, we should probably clear out the buttons for "Check for Updates"  and such before uploading to Mozilla.

**Comment by OrangutanGaming** [comment]
Yeah, #9 is probably the highest priority

**Comment by maphew** [comment]
... we should probably clear out the buttons for "Check for Updates" and
such before uploading to Mozilla.

+1

**Comment by maphew** [comment]
>
> I have a lot of experience with Discord, and think it's the perfect
> platform right now for something like this
>
Discord is new to me, but I'm willing to check it out. There's also
Gitter.im which is nice because it integrates with GitHub, auto-linking to
issues etc. I just created an instance for Lazarus so people can check it
out: https://gitter.im/lazarus-recovery/... We should settle on just one to
advertise front and centre though.

**Comment by OrangutanGaming** [comment]
I've seen a lot of people use gitter, but I'm not really a fan. There's a lot of nice features missing from it, and I don't think GitHub integration is too important

**Comment by sierracircle** [comment]
This discussion is closed. Start new threads for new issues please
